lib/cli.js

```diff
@@ -17,105 +17,176 @@
 
 const fs = require("fs"),
     path = require("path"),
-    { CLIEngine } = require("./cli-engine"),
-    options = require("./options"),
+    { promisify } = require("util"),
+    { ESLint } = require("./eslint"),
+    CLIOptions = require("./options"),
     log = require("./shared/logging"),
     RuntimeInfo = require("./shared/runtime-info");
 
 const debug = require("debug")("eslint:cli");
 
+//------------------------------------------------------------------------------
+// Types
+//------------------------------------------------------------------------------
+
+/** @typedef {import("./eslint/eslint").ESLintOptions} ESLintOptions */
+/** @typedef {import("./eslint/eslint").LintMessage} LintMessage */
+/** @typedef {import("./eslint/eslint").LintResult} LintResult */
+
 //------------------------------------------------------------------------------
 // Helpers
 //------------------------------------------------------------------------------
 
+const mkdir = promisify(fs.mkdir);
```

Comment:
Can we use the fs promises API instead?
https://nodejs.org/api/fs.html#fs_fs_promises_api

Reply:
No. We have decided the supported Node.js version is `10.12.0`, but the `fs.promises` needs `10.17.0`. See also https://github.com/eslint/rfcs/pull/44#issuecomment-546804246

Reply:
Ah ok. 👍